src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -0,0 +1,451 @@
+// Based on google, may be a problem with implicits, requiring explicit specification of some types.
+
+class SlickDataAccess(databaseConfig: Config) extends DataAccess {
+
+  def this() = this(DatabaseConfig.databaseConfig)
+
+  private val IoInput = "INPUT"
+  private val IoOutput = "OUTPUT"
+
+  import SlickDataAccess._
+
+  val dataAccess = new DataAccessComponent(databaseConfig.getString("slick.driver"))
+
+  import dataAccess.driver.api._
+
+  // TODO: Used for flatMap. Should we build another ec instead of using global?
+  private implicit val executionContext = ExecutionContext.global
+
+  // WdlValues have multiple ways they coerce back. This currently decoerces for a proper round trip.
+  // TODO: Refactor into the type?
+  private def wdlValueToString(wdlValue: WdlValue) = {
+    wdlValue match {
+      case wdlPrimitive: WdlPrimitive => wdlPrimitive.asString
+      case other => other.toString
+    }
+  }
+
+  val database = Database.forConfig("", databaseConfig)
+
+  // Check the database connection. Can be run before operations that actually use the database.
+  def isValidConnection(timeoutSecs: Int): Future[Boolean] = {
+    database.run(SimpleDBIO(_.connection.isValid(timeoutSecs))) recover { case _ => false }
+  }
+
+  // Lazily, possibly create the database, returning the result in the future
+  private lazy val createDatabase: Future[Unit] = {
+    import _root_.slick.util.ConfigExtensionMethods._
+    if (databaseConfig.getBooleanOr("slick.createSchema")) {
+      database.run(dataAccess.schema.create)
+    } else Future.successful(())
+  }
+
+  // Run action with an outer transaction. Also if we need to, create the in memory database!
+  private def runTransaction[R](action: DBIOAction[R, _ <: NoStream, _ <: Effect]): Future[R] = {
+    for {
+      _ <- createDatabase // If not created
+      result <- database.run(action.transactionally)
+    } yield result
+  }
+
+  /**
+   * Creates a row in each of the backend-info specific tables for each call in `calls` corresponding to the backend
+   * `backend`.  Or perhaps defer this?
+   */
+  override def createWorkflow(workflowInfo: WorkflowInfo,
+                              workflowInputs: Traversable[SymbolStoreEntry],
+                              calls: Traversable[Call],
+                              backend: Backend): Future[Unit] = {
+
+    val action = for {
+
+      workflowExecutionInsert <- dataAccess.workflowExecutionsAutoInc +=
+        new WorkflowExecution(
+          workflowInfo.workflowId.toString,
+          WorkflowSubmitted.toString,
+          new Date().toTimestamp)
+
+      _ <- dataAccess.workflowExecutionAuxesAutoInc += new WorkflowExecutionAux(
+        workflowExecutionInsert.workflowExecutionId.get,
+        workflowInfo.wdlSource,
+        workflowInfo.wdlJson)
+
+      symbolInsert <- dataAccess.symbolsAutoInc ++=
+        workflowInputs.toSeq map { symbol =>
```

Comment:
Could there be some intermediate locals here and below?

```
inputSymbols = workflowInputs.toSeq map { symbol => ... }
symbolInsert <- dataAccess.symbolsAutoInc ++= inputSymbols
```

I think this might particularly benefit the call/backend inserts, there's a lot going on there.  :smile: